pkg/agent/main.go

```diff
@@ -422,3 +433,32 @@ func reconcileKubelet(ctx context.Context) (bool, error) {
 	}
 	return false, nil
 }
+
+func checkManagedAgents(ctx context.Context, c *client.Client) (bool, error) {
+	args := filters.NewArgs()
+	args.Add("label", "io.cattle.agent=true")
+
+	containers, err := c.ContainerList(ctx, types.ContainerListOptions{
+		All:     true,
+		Filters: args,
+	})
+	if err != nil {
+		return false, err
+	}
+
+	for _, container := range containers {
+		con, err := c.ContainerInspect(ctx, container.ID)
+		if err != nil {
+			return false, err
+		}
+		// this check if container is a managed node-agent
+		if _, ok := container.Labels["io.kubernetes.pod.namespace"]; ok &&
```

Comment:
not sure, but do we need to also check for container state?

Reply:
I think checking the existence of the container is sufficient, don't want the state impact this logic tho.

Reply:
what's your concern?